### PR TITLE
Crypto fixes

### PR DESCRIPTION
**Pull request by ceolin**
- Ensure that sys_crand_get is getting data from entropy device only
https://github.com/zephyrproject-rtos/zephyr/issues/22223

- Fix csprng entropy initialization when using TinyCrypt

### CODE REVIEW

**Review by dleach02 — CHANGES_REQUESTED** [review]
Need to update the documentation to indicate that the sys_csrand_get() may return EIO.

**Comment by ceolin** [comment]
> Need to update the documentation to indicate that the sys_csrand_get() may return EIO.

It is already documented:
```
 * @return 0 if success, -EIO if entropy reseed error
```
Am I missing something ? I'll fix checkpatch complains though.

File: subsys/random/rand32_entropy_device.c [hunk 1]
```diff
@@ -67,6 +67,13 @@ static void rand_get(u8_t *dst, size_t outlen)
 	ret = entropy_get_entropy(dev, dst, outlen);
 
 	if (unlikely(ret < 0)) {
+		/* Don't try to fill the buffer in case of
+		 * cryptographically secure random numbers, just
+		 * propagate the driver error.
+		 */
+		if (csrand)
```
Comment (andrewboie) [inline subsys/random/rand32_entropy_device.c]: coding style requires explicit braces even for one-liners

Comment (ceolin) [reply subsys/random/rand32_entropy_device.c]: indeed, always forget to run uncrustify. Btw shouldn't it be checked by CI ?

**Comment by dleach02** [comment]
Correct, my bad

**Review by dleach02 — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**